app/helpers/automated_tests_helper.rb

```diff
@@ -378,22 +378,17 @@ def self.delete_repo(repo_dir)
   def self.scripts_to_run(assignment, call_on)
     # Find all the test scripts of the current assignment
     all_scripts = TestScript.where(assignment_id: assignment.id)
-
     list_run_scripts = Array.new
 
     # If the test run is requested at collection (by Admin or TA),
     # All of the test scripts should be run.
-    if call_on == 'collection'
-      list_run_scripts = all_scripts
-    else
-      # If the test run is requested at submission or upon request,
-      # verify the script is allowed to run.
-      all_scripts.each do |script|
-        if call_on == 'submission' && script.run_on_submission
+    # If the test run is requested at submission or upon request,
+    # verify the script is allowed to run.
+    all_scripts.each do |script|

```

Comment:
@alexvail I refactored this code in a slightly different way. When you do a pull, you'll see a conflict here; please use my version.